Userland/Services/LookupServer/LookupServer.cpp

```diff
@@ -135,7 +142,17 @@ Vector<DNSAnswer> LookupServer::lookup(const DNSName& name, DNSRecordType record
         answers.append(answer_with_original_case);
     };
 
-    // First, try local data.
+    // First, try the hostname.
+    static DNSName s_hostname = get_hostname();
```

Comment:
What if the user changes the hostname at runtime?

Reply:
That's the same problem we have with `/etc/hosts` right now. We can of course just not cache it, so let's do that. Regarding reloading `/etc/hosts`, maybe we should store the "last modified" timestamp of that in a static variable, and always check against that if we need to reload it (or we could use inode watchers, but maybe that is overkill).